### ISSUE TITLE
liblas needs C++11?

### ISSUE THREAD

**Issue by CarlosGrohmann** [OSGeo/homebrew-osgeo4mac#387]
Liblas installation failed today on a computer with OS High Sierra, Xcode 9 , CLT and a fresh install of Homebrew.

Yesterday I could install fine in one machine, today on another machine I got this error:

```
brew install liblas-gdal2 --with-laszip
==> Installing liblas-gdal2 from osgeo/osgeo4mac
==> Downloading http://download.osgeo.org/liblas/libLAS-1.8.1.tar.bz2
Already downloaded: /Users/guano/Library/Caches/Homebrew/liblas-gdal2-1.8.1.tar.bz2
==> Downloading https://github.com/libLAS/libLAS/commit/49606470.patch?full_index=1
Already downloaded: /Users/guano/Library/Caches/Homebrew/liblas-gdal2--patch-5590aef61a58768160051997ae9753c2ae6fc5b7da8549707dfd9a682ce439c8.patch
==> Patching
==> Applying 49606470.patch
patching file src/gt_citation.cpp
==> cmake .. -DWITH_GEOTIFF=ON -DWITH_GDAL=ON -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/liblas-gdal2/
==> make
Last 15 lines from /Users/guano/Library/Logs/Homebrew/liblas-gdal2/02.make:
cd /tmp/liblas-gdal2-20180607-3666-z1mtvq/libLAS-1.8.1/macbuild/src && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++  -DHAVE_GDAL=1 -DHAVE_LASZIP=1 -DHAVE_LIBGEOTIFF=1 -Dlas_EXPORTS -I/tmp/liblas-gdal2-20180607-3666-z1mtvq/libLAS-1.8.1/src/../include -I/usr/local/include -I/usr/local/opt/laszip@2.2/include -I/usr/local/opt/gdal2/include  -Wextra -Wall -Wno-unused-parameter -Wno-unused-variable -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Wredundant-decls -Wno-long-long -DNDEBUG -fPIC   -o CMakeFiles/las.dir/transform.cpp.o -c /tmp/liblas-gdal2-20180607-3666-z1mtvq/libLAS-1.8.1/src/transform.cpp
[ 20%] Building CXX object src/CMakeFiles/las.dir/utility.cpp.o
cd /tmp/liblas-gdal2-20180607-3666-z1mtvq/libLAS-1.8.1/macbuild/src && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++  -DHAVE_GDAL=1 -DHAVE_LASZIP=1 -DHAVE_LIBGEOTIFF=1 -Dlas_EXPORTS -I/tmp/liblas-gdal2-20180607-3666-z1mtvq/libLAS-1.8.1/src/../include -I/usr/local/include -I/usr/local/opt/laszip@2.2/include -I/usr/local/opt/gdal2/include  -Wextra -Wall -Wno-unused-parameter -Wno-unused-variable -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Wredundant-decls -Wno-long-long -DNDEBUG -fPIC   -o CMakeFiles/las.dir/utility.cpp.o -c /tmp/liblas-gdal2-20180607-3666-z1mtvq/libLAS-1.8.1/src/utility.cpp
1 error generated.
make[2]: *** [src/CMakeFiles/las.dir/spatialreference.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /tmp/liblas-gdal2-20180607-3666-z1mtvq/libLAS-1.8.1/src/transform.cpp:66:
In file included from /usr/local/opt/gdal2/include/gdal.h:45:
/usr/local/opt/gdal2/include/cpl_port.h:187:6: error: Must have C++11 or newer.
#    error Must have C++11 or newer.
     ^
1 error generated.
make[2]: *** [src/CMakeFiles/las.dir/transform.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/las.dir/all] Error 2
make: *** [all] Error 2

```

**Comment by nickrobison** [comment]
@CarlosGrohmann Apologies for this. We recently updated GDAL to 2.3.0, which requires C++11, I've updated the liblas formula and patched the build so that it will continue to work.

Let me know if you run into any other issues.

**Comment by CarlosGrohmann** [comment]
Thanks! Working fine now.

**Comment by gianlucalazzaro88** [comment]
Hi all,
I got a similar issue when trying to `brew install taudem`.
Here are the details:

> In file included from /tmp/taudem-20181016-51052-49gkq8/TauDEM-5.3.7/src/commonLib.h:46:
> In file included from /usr/local/opt/gdal2/include/ogr_api.h:43:
> In file included from /usr/local/opt/gdal2/include/cpl_progress.h:33:
> /usr/local/opt/gdal2/include/cpl_port.h:187:6: error: Must have C++11 or newer.
> error Must have C++11 or newer.

Anyone knows how to fix? I had tried updating/reinstalling gdal2 and liblas, but still get the error.

I recently upgraded my mac osx to Mojave, but it's my first experience with taudem so I don't know if this could depend on the os.

Thanks!
